### ISSUE TITLE
Support more than one namespace?

### ISSUE THREAD

**Issue by jwendell** [arquillian/arquillian-cube#361]
Currently (in openshift adapter) it seems we can use only one, static namespace, which must be provided upon creation time.

Is it possible to use more than one namespace?

My scenario would be: Run each class on its own namespace instead of having one namespace for the whole test suite.

We could add an observer for BeforeClass for instance, setting the namespace to be used...

**Comment by strowk** [comment]
I'd like to have even more than that - possibility to work in several namespaces in one test. I see it as being able to specify namespace like that:
    @Named(value = "test-pod", namespace = "some-namespace")

**Comment by lordofthejars** [comment]
@jwendell The question is, currently if you do not do anything Cube creates a new random `itest` namespace created by test suite execution. So what you are saying here is just create one per each test class. This has some implications in the sense that we need to create a new attribute that allows user to define the scope of namespace creation (`suite` or `class`) and update this information as well.

@strowk What you are mentioning here is having support for several already created namespaces, so what you mention here has sense but notice that in this case means that Cube is not responsible of deploying and creating all these namespaces, it is responsible of the test preparation to have everything in place, isn't it?

**Comment by jwendell** [comment]
I'm trying to remember what was my motivation to ask this at the time. Blame on me for not writing an use case in my report :/

**Comment by strowk** [comment]
@lordofthejars Yeah, in the implementation I've provided those additional namespaces are considered already existing and filled with all required resources.

We may divide the namespaces to **primary** and **secondary**. While Cube may be (if configured to) responsible for deploying of exactly one **primary** namespace, it could also inject into test example resources from any number of **secondary** namespaces if necessary, but their existence at the time of test run is considered a prerequisite as well as existence of Openshift/Kubernetess instance.
